### PR TITLE
Remove environment var `VERCEL_SPLIT_ARCHIVE` in favor of split-tgz archive option

### PR DESCRIPTION
**Pull request by onsclom**
https://github.com/vercel/vercel/pull/12671 actually adds support for splitting archives by doing `VERCEL_SPLIT_ARCHIVE=1` OR `--archive=split-tgz`, but this makes it more clear we support this through the flag.

### CODE REVIEW

File: packages/cli/src/commands/deploy/command.ts [hunk 1]
```diff
@@ -87,8 +87,7 @@ export const deployCommand = {
       type: String,
       argument: 'FORMAT',
       deprecated: false,
-      description:
-        'Compress the deployment code into a file before uploading it',
+      description: 'Compress the deployment code before uploading it',
```
Comment (TooTallNate) [inline packages/cli/src/commands/deploy/command.ts]: The text was arguably more correct before. `--archive` does not _only_ compress, but also _combines_.

File: packages/cli/src/util/deploy/process-deployment.ts [hunk 2]
```diff
@@ -97,8 +97,7 @@ export default async function processDeployment({
     vercelOutputDir,
     rootDirectory,
     skipAutoDetectionConfirmation,
-    archive:
-      process.env.VERCEL_SPLIT_ARCHIVE && archive ? 'split-tgz' : archive,
+    archive: archive,
```
Comment (TooTallNate) [inline packages/cli/src/util/deploy/process-deployment.ts]: Just to clarify, does this remove the env var support completely?

File: packages/cli/src/commands/deploy/command.ts (hunk 1 above)
Comment (onsclom) [reply packages/cli/src/commands/deploy/command.ts]: Good point, I was just avoiding how `--archive=split-tgz` compresses the deployment code into multiple files, not just one. Maybe we go with 'Compress the deployment code into an archive before uploading it'?

File: packages/cli/src/util/deploy/process-deployment.ts (hunk 2 above)
Comment (onsclom) [reply packages/cli/src/util/deploy/process-deployment.ts]: Yes. The thinking is we can tell customers to try out split archive deployments by doing `--archive=split-tgz`, and if everything goes smooth we make splitting the default `--archive=tgz` behavior.

Comment (TooTallNate) [reply packages/cli/src/util/deploy/process-deployment.ts]: I was going to say that env var would make it easier to feature flag, but for CLI running locally (which would be the case for `--archive`) we don't really have access to feature flag env vars in the same way as for deployments, so SGTM.

Comment (TooTallNate) [reply packages/cli/src/util/deploy/process-deployment.ts]:
```suggestion
    archive,
```

**Review by EndangeredMassa — APPROVED** [review]

**Review by trek — APPROVED** [review]

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**